I encountered the same issue as #39789  x64-linux, I can submit a bug report if you wish.
As reported
- the same issue was reported with regards to the abseil port
- a fix PR was published (but not yet merged) for abseil  #39803
- this current PR for google-cloud-cpp #39802 
- I applied both to my vcpkg repo and soldiered on
```
 git pull origin pull/39803/head
 git pull --rebase=merges  origin pull/39802/head
```
- I still get similar errors for some of the libraries
```
CMake Error at google/cloud/google_cloud_cpp_common.cmake:294 (target_link_libraries):
  The link interface of target "google_cloud_cpp_mocks" contains:

    GTest::gmock_main

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  google/cloud/CMakeLists.txt:53 (include)


CMake Error at cmake/GoogleCloudCppLibrary.cmake:350 (target_link_libraries):
  The link interface of target "google_cloud_cpp_monitoring_mocks" contains:

    GTest::gmock_main

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  google/cloud/monitoring/CMakeLists.txt:19 (google_cloud_cpp_add_gapic_library)


CMake Error at cmake/GoogleCloudCppLibrary.cmake:350 (target_link_libraries):
  The link interface of target "google_cloud_cpp_trace_mocks" contains:

    GTest::gmock_main

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  google/cloud/trace/CMakeLists.txt:19 (google_cloud_cpp_add_gapic_library)


CMake Error at cmake/GoogleCloudCppLibrary.cmake:350 (target_link_libraries):
  The link interface of target "google_cloud_cpp_iam_mocks" contains:

    GTest::gmock_main

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  google/cloud/iam/CMakeLists.txt:19 (google_cloud_cpp_add_gapic_library)


CMake Error at cmake/GoogleCloudCppLibrary.cmake:350 (target_link_libraries):
  The link interface of target "google_cloud_cpp_bigquery_mocks" contains:

    GTest::gmock_main

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  google/cloud/bigquery/bigquery_grpc.cmake:19 (google_cloud_cpp_add_gapic_library)
  google/cloud/bigquery/CMakeLists.txt:20 (include)


CMake Error at google/cloud/bigtable/CMakeLists.txt:340 (target_link_libraries):
  The link interface of target "google_cloud_cpp_bigtable_mocks" contains:

    GTest::gmock_main

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

CMake Error at cmake/GoogleCloudCppLibrary.cmake:350 (target_link_libraries):
  The link interface of target "google_cloud_cpp_iap_mocks" contains:

    GTest::gmock_main

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  google/cloud/iap/CMakeLists.txt:19 (google_cloud_cpp_add_gapic_library)


CMake Error at cmake/GoogleCloudCppLibrary.cmake:350 (target_link_libraries):
  The link interface of target "google_cloud_cpp_kms_mocks" contains:

    GTest::gmock_main

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  google/cloud/kms/CMakeLists.txt:19 (google_cloud_cpp_add_gapic_library)


CMake Error at cmake/GoogleCloudCppLibrary.cmake:350 (target_link_libraries):
  The link interface of target "google_cloud_cpp_logging_mocks" contains:

    GTest::gmock_main

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  google/cloud/logging/CMakeLists.txt:19 (google_cloud_cpp_add_gapic_library)


CMake Error at google/cloud/pubsub/CMakeLists.txt:394 (target_link_libraries):
  The link interface of target "google_cloud_cpp_pubsub_mocks" contains:

    GTest::gmock_main

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at google/cloud/spanner/CMakeLists.txt:374 (target_link_libraries):
  The link interface of target "google_cloud_cpp_spanner_mocks" contains:

    GTest::gmock_main

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at cmake/GoogleCloudCppLibrary.cmake:350 (target_link_libraries):
  The link interface of target "google_cloud_cpp_speech_mocks" contains:

    GTest::gmock_main

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.
Call Stack (most recent call first):
  google/cloud/speech/CMakeLists.txt:19 (google_cloud_cpp_add_gapic_library)


CMake Error at cmake/GoogleCloudCppLibrary.cmake:350 (target_link_libraries):
  The link interface of target "google_cloud_cpp_storageinsights_mocks"
  contains:

    GTest::gmock_main

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  google/cloud/storageinsights/CMakeLists.txt:19 (google_cloud_cpp_add_gapic_library)


CMake Error at cmake/GoogleCloudCppLibrary.cmake:350 (target_link_libraries):
  The link interface of target "google_cloud_cpp_storagetransfer_mocks"
  contains:

    GTest::gmock_main

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  google/cloud/storagetransfer/CMakeLists.txt:29 (google_cloud_cpp_add_gapic_library)


CMake Error at cmake/GoogleCloudCppLibrary.cmake:350 (target_link_libraries):
  The link interface of target "google_cloud_cpp_support_mocks" contains:

    GTest::gmock_main

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  google/cloud/support/CMakeLists.txt:19 (google_cloud_cpp_add_gapic_library)


CMake Error at cmake/GoogleCloudCppLibrary.cmake:350 (target_link_libraries):
  The link interface of target "google_cloud_cpp_tasks_mocks" contains:

    GTest::gmock_main

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  google/cloud/tasks/CMakeLists.txt:19 (google_cloud_cpp_add_gapic_library)


CMake Error at cmake/GoogleCloudCppLibrary.cmake:350 (target_link_libraries):
  The link interface of target "google_cloud_cpp_translate_mocks" contains:

    GTest::gmock_main

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  google/cloud/translate/CMakeLists.txt:19 (google_cloud_cpp_add_gapic_library)


CMake Warning:
  Manually-specified variables were not used by the project:

    FETCHCONTENT_FULLY_DISCONNECTED
    VCPKG_PLATFORM_TOOLSET
    VCPKG_SET_CHARSET_FLAG
    _VCPKG_ROOT_DIR


CMake Generate step failed.  Build files cannot be regenerated correctly.
```

the additional targets are due to the fact that I build with a vcpkg like
```json
{
    "dependencies" :[
        "abseil",
        {
            "name" : "google-cloud-cpp",
            "features" : [
                "grpc-common",
                "iam",
                "iap",
                "compute",
                "kms",
                "monitoring",
                "logging",
                "opentelemetry",
                "pubsub",
                "rest-common" ,
                "speech",
                "storage",
                "storageinsights",
                "storagetransfer",
                "support",
                "tasks",
                "trace",
                "translate"
            ]
        }
   ]
}
```

but it looks like the `-DBUILD_TESTING=OFF -DGOOGLE_CLOUD_CPP_WITH_MOCKS=OFF` flags do not make it to my CMakeCache.txt ....

--- a/ports/google-cloud-cpp/fix_mocks_dependent_option.patch
+++ b/ports/google-cloud-cpp/fix_mocks_dependent_option.patch
@@ -1,0 +1,22 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index e1a1dc8..807f6e2 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -173,7 +173,7 @@ provided mocks to test code involving the Cloud C++ clients may wish to turn
+ this flag off.]==]
+     ON
+     "NOT BUILD_TESTING"
+-    ON)
++    $<IF:$<BOOL:${BUILD_TESTING}>:OFF:ON>)
+ mark_as_advanced(GOOGLE_CLOUD_CPP_WITH_MOCKS)
+ 
+ # The examples use exception handling to simplify the code. Therefore they
+@@ -246,7 +246,7 @@ include(CTest)
+ # used in the depends condition of the next option.
+ include(EnableCxxExceptions)
+ 
+-if (BUILD_TESTING)
++if (BUILD_TESTING OR GOOGLE_CLOUD_CPP_WITH_MOCKS)
+     # Discover and add targets for the GTest::gtest and GTest::gmock libraries.
+     include(FindGMockWithTargets)
+ endif ()

--- a/ports/google-cloud-cpp/portfile.cmake
+++ b/ports/google-cloud-cpp/portfile.cmake
@@ -8,6 +8,7 @@ vcpkg_from_github(
     HEAD_REF main
     PATCHES
         support_absl_cxx17.patch
+        fix_mocks_dependent_option.patch
 )
 
 if ("grpc-common" IN_LIST FEATURES)

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -3154,7 +3154,7 @@
     },
     "google-cloud-cpp": {
       "baseline": "2.26.0",
-      "port-version": 0
+      "port-version": 1
     },
     "google-cloud-cpp-common": {
       "baseline": "alias",

--- a/versions/g-/google-cloud-cpp.json
+++ b/versions/g-/google-cloud-cpp.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "79809a3d4f368d9afe5eac6f10d892311b464e00",
+      "version": "2.26.0",
+      "port-version": 1
+    },
+    {
       "git-tree": "6f92046f144ea5804abafe724ea2fec115dad148",
       "version": "2.26.0",
       "port-version": 0

